### PR TITLE
Fix error when trying to delete an image due to a bad container

### PR DESCRIPTION
**Pull request by duglin**
I ran into a situation where I was trying:
    `docker rmi busybox`
and it kept failing saying:
    `could not find image: Prefix can't be empty`

While I have no idea how I got into this situation, it turns out this is
error message is from `daemon.canDeleteImage()`. In that func we loop over
all containers checking to see if they're using the image we're trying to
delete.  In my case though, I had a container with no ImageID. So the code
would die trying to find that image (hence the "Prefix can't be empty" err).
This would stop all processing despite the fact that the container we're
checking had nothing to do with 'busybox'.

My change logs the bad situation in the logs and then skips that container.
There's no reason to fail all `docker rmi ...` calls just because of one
bad container.

Will continue to try to figure out how I got a container w/o an ImageID
but as of now I have no idea, I didn't do anything but normal docker cli
commands.

Signed-off-by: Doug Davis dug@us.ibm.com

### CODE REVIEW

File: daemon/image_delete.go [hunk 1]
```diff
@@ -140,6 +141,16 @@ func (daemon *Daemon) imgDeleteHelper(name string, list *[]types.ImageDelete, fi
...
 func (daemon *Daemon) canDeleteImage(imgID string, force bool) error {
 	for _, container := range daemon.List() {
+		if container.ImageID == "" {
+			// This technically should never happen, but if the container
+			// has no ImageID then log the situation and move on.
+			// If we allowed processing to continue then the code later
+			// on would fail with a "Prefix can't be empty" error even
+			// though the bad container has nothing to do with the image
+			// we're trying to delete.
+			logrus.Errorf("Container %q has no image associted with it!", container.ID)
```
Comment (tiborvass) [inline daemon/image_delete.go]: `associated`

Comment (duglin) [inline daemon/image_delete.go]: fixed- thanks!

**Comment by duglin** [comment]
In case people are wondering why this situation isn't a problem in other areas of the code, it actually is. While debugging I found several other flows where this error is generated, but in each case the error was eventually ignored and processing continued. So, in that respect this fix is consistent with those other cases I came across.

**Comment by calavera** [comment]
LGTM

**Comment by duglin** [comment]
ping @jfrazelle @LK4D4

**Comment by jfrazelle** [comment]
is this for 1.7

**Comment by duglin** [comment]
@jfrazelle I haven't heard of other people complaining about seeing this error so I don't think its critical for 1.7, despite it being very safe.

**Comment by jfrazelle** [comment]
LGTM

**Pull request merged**